### DISCUSSION TITLE
BiblioteQ's Features

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#114] [General]
Hello
I need to study your software to see what features they offer, 

I'm looking at software like MesLivrePro , Tellico and Gcstar and then BiblioteQ.
The comparison is interesting because some functions are great on 1 and not so good on the other and vice versa. The ideal will be a fusion of the best of 4 :-)

for example the management of the csv import is from my point of view "really difficult to understand" compared to gcstar and tellico.
For the management of users, BibloteQ is far ahead

I am ready to invest in BiblioteQ to make tests (I managed to compile it with qtcreator ... and give tracks but for our choice we are looking for a software that provides the maximum of the following: 

Unimarc import (Unimarc with field 995)
Batch import .CSV
Batch record deletion (BDP return)
Batch export .CSV
User management
User Management Import .csv
User management Export to .cvs
Backup of the database simply / automatically (basic user)
Contribution management
Loans management with handheld
Returns management with handheld
Loans management with internal barcode
Returns management with internal barcode
Loans in progress report

Question: what function(s) do you expect to add by June 2022 when we choose our software.

I am attaching my progress report
[rapport-p3.pdf](https://github.com/textbrowser/biblioteq/files/7469007/rapport-p3.pdf)

Thank you in advance

**Comment by textbrowser** [comment]
Yikes, that's a lot of import work. As you can see for yourself, most of the work is translating (importing) some legacy formats into a new system. As for handheld devices, Android probably offers free barcode scanners. I just released the first Android version. It does not include PQ (PostgreSQL) because the driver is not readily available; so I have to find time for it.

I would categorize the following as important:
- a means of presenting more views (in non-Android this is possible by multiple BQ instances, however, it would be very helpful if one could view several categories simultaneously)
- an improved mechanism for the reservation process
- an import mechanism exists; it requires more templates
- desktop integration with android devices, perhaps through BT or TCP

BQ does not have a team, is fully open source, and is essentially a hobby. :) As a hobby, it can always be expanded.

**Comment by textbrowser** [comment]
PostgreSQL includes backup information so I'll leave that for the administrators. For SQLite, these are files and best practices are left for the owners of the data. Therefore, data backup is not an issue.

The current import is modeled after Libre Office. It's programmable in the sense that you have to describe your data.
1 | 2 | 3 | ... | n
Each row of your import file has specific columns. You being the owner know that column 1, let's say, references book ISBNs, and so on. There is a template already. If you have specific import requirements, please include them. Or, improved import mechanisms.

**Comment by meteos77** [comment]
I thank you for your quick answers and I understand that this is just a hobby.

our system must run on windows10 (library computer) with a handheld. if we choose BiblioteQ we will be with the SQLite version because no BDD admin and not very powerful pc.(AMD Athlon II X2 250u)
.
for the moment I see 2 blocking points on BiblioteQ for our use,
1 - the unimarc import from a file that can contain 300 records with management of fields 995 which should add some fields to the database
for example 
995 $a internal barcode of the central library
995 $m (date of the loan from the central library)
995 $n (expected date of return to the central library)
call number, target audience ...
...
2- the management of the loans which seems to me to be book by book ? 

I can of course provide you with the most accurate way of how I see the integration of the features (with my past experiences).


at the moment I am translating the doc in my language (French)

best regards

**Comment by textbrowser** [comment]
What's a BDD admin? Why not PostgreSQL? It will work on the Windows 10 machine, that is, a local instance. It far surpasses SQLite and if you decide to expand the network, the database will be accessible by more computers. SQLite is for individuals.

The issues that you mention are issues of resources (time). They're not difficult to implement. Schema changes are easy and I provide update processes for both database engines.

Yes, reservation is available for every category save for photographs. And yes, per item.

**Reply by meteos77** [reply]
sorry it's french, BDD = database administrator

**Reply by meteos77** [reply]
I bounced on And yes, per item.

as far as I understand the way the software works at the moment, I look for the book, I select the book and I assign it to the person.

I would prefer, I select the person, I scan (the ISBN or the Bar Code) with the shower head all the books 1 by 1 and the software places them in its "loan".
knowing that I scan 1 book in 5 seconds ... the person borrows the 20 books in maximum 1 or 2 minutes.

**Reply by textbrowser** [reply]
I can implement your preferred approach also.

**Comment by textbrowser** [comment]
Of course, PQ includes a large community, backup best practices, Docker support, can exist in a NAS, offers more security, growth, etc. SQLite is for petite projects.

**Reply by meteos77** [reply]
we are a library of volunteers
about 50 members who borrow books
and 8 - 10 people volunteer to run the library
about 10000 books in the library

no internet access in the library
no technical staff (the volunteers are rather teachers, accountant... no computer technician except me)

every 3 months 
300 books go back to the central library
300 books arrive from the central library

our needs are not huge
one pc does everything
no computer consultation of available books

**Reply by meteos77** [reply]
for you according to the data and is it a small structure

**Reply by textbrowser** [reply]
You don't need Internet access for PQ. You can create a small network in the building and allow 2 or more people to access the database from multiple computers. Multiple people can browse and/or reserve items. PQ has a nice installation guide for Windows.

**Comment by textbrowser** [comment]
There are also lots of tools for PQ as well as a command-line interface. So, you can write a Web interface later and if you decide to move from BQ you can export the schema (PQ includes export and import processes).

**Comment by textbrowser** [comment]
PQ also includes actual accounts unlike SQLite.

**Comment by textbrowser** [comment]
I've also tested PQ on a Pi 4. It's functional. I would not recommend using it for an enterprise, but the fact that it works well on a Pi clearly shows the strength of PQ. I'm not a spokesperson. It's open source too.

**Reply by meteos77** [reply]
I do not doubt the superiority of pg over sqlite
I used it with the OpenStreetMap project (also Opensource)  a few years ago
